var/spack/repos/builtin/packages/cscope/package.py

```diff
@@ -10,8 +10,15 @@ class Cscope(AutotoolsPackage):
     """Cscope is a developer's tool for browsing source code."""
 
     homepage = "http://cscope.sourceforge.net/"
-    url      = "http://downloads.sourceforge.net/project/cscope/cscope/15.8b/cscope-15.8b.tar.gz"
 
+    def url_for_version(self, version):
+        if '@:15.8b' in self.spec:
+            url = "https://sourceforge.net/projects/cscope/files/cscope/v15.9/cscope-15.9.tar.gz"
+        else:
+            url = "https://sourceforge.net/projects/cscope/files/cscope/v15.9/cscope-15.9.tar.gz"
+        return url
```

Comment:
This doesn't make any sense. What are you trying to do here?

Reply:
Just trying to adjust the download URL. The download schemes are different and spack doesn't get the right version if I just insert one URL... I guess this is a workaround inspired by [the documentation](https://spack.readthedocs.io/en/latest/packaging_guide.html#versions-and-fetching)... since I'm pretty new to spack and python I'm really open for a better solution :)

Reply:
It looks to me like this function returns the same string regardless of which version is needed. I think what you're looking for is:
```python
def url_for_version(self, version):
    url = "https://sourceforge.net/projects/cscope/files/cscope/{0}{1}/cscope-{1}.tar.gz"
    if version >= Version('15.9')
        return url.format('v', version)
    else:
        return url.format('', version)
```
You can confirm whether or not the URLs are correct by running `spack info cscope`